systemtest/src/test/java/io/strimzi/systemtest/operators/FeatureGatesST.java

```diff
@@ -231,27 +242,39 @@ void testSwitchingConnectStabilityIdentifiesFeatureGateOnAndOff(ExtensionContext
         Map<String, String> connectPods = PodUtils.podSnapshot(clusterOperator.getDeploymentNamespace(), connectLabelSelector);
+        String startingConnectorPodName = kubeClient().listPods(testStorage.getNamespaceName(), connectLabelSelector).get(0).getMetadata().getName();
 
-        KafkaClients clients = new KafkaClientsBuilder()
-            .withProducerName(testStorage.getProducerName())
+        // We are sending continuous messages throughout the test to verify communication in between enabling the FG
+        KafkaClients continuousClients = new KafkaClientsBuilder()
+            .withProducerName("continuous-producer")
             .withConsumerName(testStorage.getConsumerName())
             .withBootstrapAddress(KafkaResources.plainBootstrapAddress(testStorage.getClusterName()))
             .withTopicName(testStorage.getTopicName())
-            .withMessageCount(messageCount)
+            .withMessageCount(continuousMessageCount)
+            .withMessage(continuousMessage)
             .withDelayMs(500)
             .withNamespaceName(clusterOperator.getDeploymentNamespace())
             .build();
 
-        String connectorPodName = kubeClient().listPods(testStorage.getNamespaceName(), connectLabelSelector).get(0).getMetadata().getName();
-
-        // we are sending messages continuously throughout the test to check that connector is working
-        resourceManager.createResourceWithWait(extensionContext, clients.producerStrimzi());
+        // We also want to check simple communication in each step
+        KafkaClients oneTimeClients = new KafkaClientsBuilder()
+            .withProducerName(startingProducerName)
+            .withConsumerName(testStorage.getConsumerName())
+            .withBootstrapAddress(KafkaResources.plainBootstrapAddress(testStorage.getClusterName()))
+            .withTopicName(testStorage.getTopicName())
+            .withMessageCount(oneTimeMessageCount)
+            .withMessage(startingMessage)
+            .withNamespaceName(clusterOperator.getDeploymentNamespace())
+            .build();
 
-        LOGGER.info("Verifying that KafkaConnector is able to sink the messages to the file-sink file");
-        KafkaConnectUtils.waitForMessagesInKafkaConnectFileSink(testStorage.getNamespaceName(), connectorPodName, Constants.DEFAULT_SINK_FILE_PATH, "Hello-world");
+        LOGGER.info("Verifying that KafkaConnector is able to sink the messages to the file-sink file before enabling FG");
+        resourceManager.createResourceWithWait(extensionContext, continuousClients.producerStrimzi());
+        resourceManager.createResourceWithWait(extensionContext, oneTimeClients.producerStrimzi());
+        ClientUtils.waitForClientSuccess(startingProducerName, testStorage.getNamespaceName(), oneTimeMessageCount);
+        KafkaConnectUtils.waitForMessagesInKafkaConnectFileSink(testStorage.getNamespaceName(), startingConnectorPodName, Constants.DEFAULT_SINK_FILE_PATH, startingMessage);
```

Comment:
For the "one time" clients you can wait for the particular message to appear (so `message - 49`) right?